src/NoSingleLineRecords.elm

```diff
@@ -0,0 +1,103 @@
+module NoSingleLineRecords exposing (rule)
+
+{-|
+
+@docs rule
+
+-}
+
+import Elm.Syntax.Declaration as Declaration exposing (Declaration)
+import Elm.Syntax.Node as Node exposing (Node(..))
+import Elm.Syntax.Range as Range exposing (Range)
+import Elm.Syntax.TypeAnnotation as TypeAnnotation exposing (TypeAnnotation)
+import Review.Fix as Fix exposing (Fix)
+import Review.Rule as Rule exposing (Rule)
+
+
+{-| Reports record definitions in type aliases that are written on a single line.
+
+    config =
+        [ NoSingleLineRecords.rule
+        ]
+
+
+## Fail
+
+    type alias Person =
+        { name : String, age : Int, address : Address }
+
+
+## Success
+
+    type alias Person =
+        { name : String
+        , age : Int
+        , address : Address
+        }
+
+
+## When (not) to enable this rule
+
+You don't ever need to enable this rule. This is a code style that can easily be reviewed manually as appropriate.
+
+
+## Try it out
+
+You can try this rule out by running the following command:
+
+```bash
+elm-review --template sparksp/elm-review-rules-to-avoid/example --rules NoSingleLineRecords
+```
+
+-}
+rule : Rule
+rule =
+    Rule.newModuleRuleSchema "NoSingleLineRecords" ()
+        |> Rule.withSimpleDeclarationVisitor declarationVisitor
+        |> Rule.fromModuleRuleSchema
+
+
+declarationVisitor : Node Declaration -> List (Rule.Error {})
+declarationVisitor declaration =
+    case Node.value declaration of
+        Declaration.AliasDeclaration { typeAnnotation } ->
+            aliasDeclarationVisitor typeAnnotation
+
+        _ ->
+            []
+
+
+aliasDeclarationVisitor : Node TypeAnnotation -> List (Rule.Error {})
+aliasDeclarationVisitor (Node range typeAnnotation) =
+    case ( typeAnnotation, linesInRange range ) of
+        ( TypeAnnotation.Record _, 1 ) ->
```

Comment:
I don't know if `elm-analyse` does it too, but you probably need to report the issue for `TypeAnnotation.GenericRecord` too

Reply:
I don't know if it does, but it probably makes sense as you say. Will be easy to add.